translation/message_selector_test.go

```diff
@@ -0,0 +1,205 @@
+package translation
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/suite"
+)
+
+type MessageSelectorTestSuite struct {
+	suite.Suite
+	selector *MessageSelector
+}
+
+func TestMessageSelectorTestSuite(t *testing.T) {
+	suite.Run(t, new(MessageSelectorTestSuite))
+}
+
+func (m *MessageSelectorTestSuite) SetUpTest() {
+	m.selector = NewMessageSelector()
+}
+
+func (m *MessageSelectorTestSuite) TestChoose() {
+	tests := []struct {
+		expected string
+		message  string
+		number   int
+		locale   string
+	}{
+		{"first", "first", 0, "en"},
+		{"first", "first", 10, "en"},
+		{"first", "first|second", 1, "en"},
+		{"second", "first|second", 10, "en"},
+		{"second", "first|second", 0, "en"},
+
+		{"first", "{0} first|{1}second", 0, "en"},
+		{"first", "{1}first|{2}second", 1, "en"},
+		{"second", "{1}first|{2}second", 2, "en"},
+		{"first", "{2}first|{1}second", 2, "en"},
+		{"second", "{9}first|{10}second", 0, "en"},
+		{"first", "{9}first|{10}second", 1, "en"},
+		{"", "{0}|{1}second", 0, "en"}, // there is some problem with it
+		{"", "{0}first|{1}", 1, "en"},  // same
+		{"first\nline", "{1}first\nline|{2}second", 1, "en"},
+		{"first \nline", "{1}first \nline|{2}second", 1, "en"},
+		{"first", "{0}first|[1,9]second", 0, "en"},
+		{"second", "{0}first|[1,9]second", 1, "en"},
+		{"second", "{0}first|[1,9]second", 10, "en"},
+		{"first", "{0}first|[2,9]second", 1, "en"},
+		{"second", "[4,*]first|[1,3]second", 1, "en"},
+		{"first", "[4,*]first|[1,3]second", 100, "en"},
+		{"second", "[1,5]first|[6,10]second", 7, "en"},
+		{"first", "[*,4]first|[5,*]second", 1, "en"},
+		{"second", "[5,*]first|[*,4]second", 1, "en"},
+		{"second", "[5,*]first|[*,4]second", 0, "en"},
+
+		{"first", "{0}first|[1,3]second|[4,*]third", 0, "en"},
+		{"second", "{0}first|[1,3]second|[4,*]third", 1, "en"},
+		{"third", "{0}first|[1,3]second|[4,*]third", 9, "en"},
+
+		{"first", "first|second|third", 1, "en"},
+		{"second", "first|second|third", 9, "en"},
+		{"second", "first|second|third", 0, "en"},
```

Comment:
Q: How to understand this case?

Reply:
![image](https://github.com/goravel/framework/assets/84431594/6db14ca1-719b-4163-8722-67c746f4c26c)
if locale is `en` and number == 1 then only it will return value of index=1 else it will return value of index=0.